File: Python/services/file_reader.py
from utils.csv_reader import CsvReader
from utils.json_reader import JsonReader
from services.create_insert import create_insert


class FileReader:
    def __init__(self, flag: str):
        if flag == "--c":
            self.reader: CsvReader = CsvReader()
            self.reader.read_notes()
        else:
            self.reader: JsonReader = JsonReader()
            self.reader.read_notes()
        create_insert(self.reader.notes)

File: Go/services/file_reader.go
package services

import (
	"fmt"

	"github.com/siddhant-vij/Note-Taking-App/models"
	"github.com/siddhant-vij/Note-Taking-App/utils"
)

type Reader interface {
	ReadNotes() error
	Notes() []models.Note
}

func NewFileReader(flag string, uri string) error {
	var r Reader
	switch flag {
	case "--c":
		r = &utils.CSVReader{}
	case "--j":
		r = &utils.JSONReader{}
	default:
		return fmt.Errorf("unsupported flag: %s", flag)
	}
	err := r.ReadNotes()
	if err != nil {
		return err
	}
	CreateInsert(uri, r.Notes())
	return nil
}
